### PR TITLE
Feat/routes crud

### PR DESCRIPTION
**Pull request by ntwari-egide**
## What does this PR do

-   It shows implementation of Routes model, service, controller and routes.

## Description

- As  **an operator**, I want to create routes so that buses can travel in pre-defined routes.

- As  **an operator**, I want to create routes so that users can view in pre-defined routes.

## How to manually test this

-   Check in the branch  `feat/routes-crud`
-   Run the server
-   Open the Swagger API documentation to test the buses

## Relevant Trello story

[7-crud-operations-for-routes][(https://trello.com/c/siqznvua/7-crud-operations-for-routes)](https://trello.com/c/siqznvua/7-crud-operations-for-routes)

## Screenshots

enter image description here

### CODE REVIEW

File: src/api/controllers/routes.controller.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const route = updateRoute(req.params.id, req.body);
+    
+    res.send(route);
+}
+
+export const routeDelete = (req, res) => {
+    const route = deleteRoute(req.params.id);
+    
+    res.send(route);
+}
```
Comment (TuyizeeAnastase) [inline src/api/controllers/routes.controller.js]: i need you send json files [refer](https://github.com/atlp-rwanda/rca-phantom-team1-bn/pull/10/files#diff-259e2aeb7411d1bdb2045e475b436e668eff745d5fe360df7bd5274d1c927091) to this

File: src/api/routes/routes.routes.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+ *
+ *     responses:
+ *       200:
+ *         description: The route was deleted
+ *       404:
+ *         description: The route was not found
+ */
+
+router.delete("/:id", routeDelete);
+
```
Comment (TuyizeeAnastase) [inline src/api/routes/routes.routes.js]: add middleware to check if route exist by `id`  before editing or deleting

File: src/api/routes/routes.routes.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+ *       200:
+ *         description: The route was successfully created
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Route'
+ *       500:
+ *         description: Some server error
+ */
+
```
Comment (TuyizeeAnastase) [inline src/api/routes/routes.routes.js]: add middleware to check if route with that id exist

File: src/api/routes/routes.routes.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+// eslint-disable-next-line consistent-return
+router.post("/", saveRoute);
+
+/**
```
Comment (TuyizeeAnastase) [inline src/api/routes/routes.routes.js]: - middleware to validate user input  check [this ](https://github.com/atlp-rwanda/rca-phantom-team1-bn/pull/10/files#diff-c62c99e58d96d62ea668b8e8477a2c1272ad5e2f946a31cf98e7bf4c9f647669)
- add middleware to check if router not already exist

File: src/index.js [hunk 5]
```diff
@@ -65,6 +68,9 @@ app.use(express.json());
 app.use(morgan("dev"));
 app.use(appRouter);
 
+app.use("/buses", busesRouter);
+app.use("/routes", routesRouter);
```
Comment (TuyizeeAnastase) [inline src/index.js]: add permissions only for operator ask Patrick for assistince

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]

**Review by thepatrickniyo — COMMENTED** [review]
There is an error in installing dependencies as  I can see on GitHub actions. The error is possibly due to collapsed `package-lock.json` or a library with an issue, you installed. 
If you have time we can look into it.

**Pull request closed without merging**